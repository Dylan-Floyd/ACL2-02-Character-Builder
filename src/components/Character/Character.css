.char-img-div {
  width: 150px;
  height: 150px;
  background-size: cover;
  background-repeat: no-repeat;
}
