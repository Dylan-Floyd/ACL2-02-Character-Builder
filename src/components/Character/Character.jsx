import React from 'react'

import './Character.css'

export default function Character({
  head,
  torso,
  legs
}) {
  return (
    <div>
      <div
        aria-label='head'
        className='char-img-div'
        style={{ backgroundImage: `url(./${head}-head.png)` }}
      />
      <div
        aria-label='torso'
        className='char-img-div'
        style={{ backgroundImage: `url(./${torso}-middle.png)` }}
      />
      <div
        aria-label='legs'
        className='char-img-div'
        style={{ backgroundImage: `url(./${legs}-pants.png)` }}
      />
    </div>
  )
}
