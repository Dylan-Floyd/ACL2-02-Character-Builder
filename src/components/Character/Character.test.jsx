import { screen, render } from '@testing-library/react'
import Character from './Character.jsx'

it('should render the character component', () => {
  const { container } = render(<Character head='dog' torso='blue' legs='leg'/>)
  const head = screen.getByLabelText(/head/)
  const torso = screen.getByLabelText(/torso/)
  const legs = screen.getByLabelText(/legs/)

  expect(head).toHaveStyle('backgroundImage: url(./dog-head.png)')
  expect(torso).toHaveStyle('backgroundImage: url(./blue-middle.png)')
  expect(legs).toHaveStyle('backgroundImage: url(./leg-pants.png)')
  expect(head).toBeInTheDocument()
  expect(torso).toBeInTheDocument()
  expect(legs).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})
