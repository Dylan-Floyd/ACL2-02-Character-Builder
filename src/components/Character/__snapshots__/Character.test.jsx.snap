// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the character component 1`] = `
<div>
  <div>
    <div
      aria-label="head"
      class="char-img-div"
      style="background-image: url(./dog-head.png);"
    />
    <div
      aria-label="torso"
      class="char-img-div"
      style="background-image: url(./blue-middle.png);"
    />
    <div
      aria-label="legs"
      class="char-img-div"
      style="background-image: url(./leg-pants.png);"
    />
  </div>
</div>
`;
